### PR TITLE
[13.0][IMP] hr_employee_calendar_planning: Allow duplicate employee

### PR DESCRIPTION
**Pull request by victoralmau**
Allow duplicate employee.

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT41700

### CODE REVIEW

File: hr_employee_calendar_planning/models/hr_employee.py [hunk 1]
```diff
@@ -113,6 +114,15 @@ def regenerate_calendar(self):
         for item in self:
             item._regenerate_calendar()
 
+    def copy(self, default=None):
+        self.ensure_one()
+        new = super().copy(default)
+        # We define the first as the main calendar
+        new.resource_id.calendar_id = fields.first(new.calendar_ids).calendar_id
```
Comment (pedrobaeza) [inline hr_employee_calendar_planning/models/hr_employee.py:121]: Can we avoid this? We are regenerating after the main calendar, and it's too late for filling the required field. I'm also doubting if the regenerate call is needed.

Comment (victoralmau) [reply hr_employee_calendar_planning/models/hr_employee.py:121]: I have tried other possibilities and this is the "_least hard_".

Maybe another option would be that `calendar_id` field of `resource.resource` is not copy, do you want me to investigate a bit more here?

Comment (pedrobaeza) [reply hr_employee_calendar_planning/models/hr_employee.py:121]: OK, if that's the reason, then please clarify it with a good comment. The current one is not clarifying anything. If I'm understanding OK the problem, the comment would be something like `# Define a good main calendar for being able to regenerate it later`.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@yajo maybe this solves your problem in #1167

**Review by chienandalu — APPROVED** [review]

**Comment by yajo** [comment]
I don't think so. That problem was happening when installing an addon's demo data, but code here only affects when copying other employees AFAICS.

**Comment by pedrobaeza** [comment]
OK, let's merge this one and fw-port it, and we'll continue checking edge cases:

/ocabot merge patch

**Pull request merged**